### ISSUE TITLE
Issues with queue - Tracks on end of queue deleted / No deletions shown / forked-daapd crashes

### ISSUE THREAD

**Issue by knedlsepp** [ejurgensen/forked-daapd#160]
I am using the _Remote for Windows Phone_ together with a Raspberry Pi 2 running forked-daapd.
The Windows Phone App works as expected when connected to an actual iTunes library on a Mac, but shows the following buggy behavior when connected to forked-daapd on the Pi:

When I start playback of an entire album and then browse the titles in the resulting queue and tap one of them to skip forward to this track, tracks from the back of the queue get deleted. (I click the first element in the queue, which starts playing and correctly gets deleted from the queue, but at the same time the last item in the queue also gets deleted. Here are three examples:
Queue: [1,2,3,4,5,6]. Click on [1]. Resulting queue: [2,3,4,5]. 
Queue: [1,2,3,4,5,6]. Click on [2]. Resulting queue: [3,4].
Queue: [1,2,3,4,5,6]. Click on [3]. Resulting queue: [].
) 
I would expect only the tracks up to the tapped item to be deleted. The corresponding log entry is: 

```
[2015-05-29 00:54:32] [DEBUG]     dacp: DACP request: /ctrl-int/1/playqueue-edit?command=playnow&index=2&session-id=998701246
```

In a similar story, when I try to delete one of the tracks from the queue by clicking on the bin icon, the queue in _Remote for Windows Phone_ does not change at all. I would expect the entry to disappear instead. The corresponding log entry is:

```
[2015-05-29 00:56:22] [DEBUG]     dacp: DACP request: /ctrl-int/1/playqueue-edit?command=remove&items=1&session-id=998701246
[2015-05-29 00:56:22] [DEBUG]   player: Removing item from position 1
```

It seems forked-daapd somehow deletes it internally though. When I click on the bin button of the second entry and then start the first entry in the queue, the second entry will be deleted afterwards.
When I repeatedly press the bin button of the first track more times than there are elements in the queue, forked-daapd crashes.

**Comment by chme** [comment]
I cannot reproduce the error "Tracks on end of queue deleted" with the android remote app Retune. forked-daapd should behave like you expect. Unfortunately i do not have a windows phone to see where the problem lies. Maybe a debug log could help shed some light into this.

Regarding the "No deletions shown" issue, my guess is that the app relies on an update notification, that should be send by forked-daapd after removing the item from the queue. I will try to see what iTunes does after removing/modifying the queue.
The crash when deleting an item not in the queue should be fixed by #166.

**Comment by knedlsepp** [comment]
@chme Great to hear of the fix #166. What could I do to narrow down the issue? Save the forked-daapd log for the deletion and sniff the http-packets?
